### ISSUE TITLE
Add dynamic hooks to a dynamic consent system such as CTRL

### ISSUE THREAD

**Issue by andrewpatto** [umccr/elsa-data#9]
During the phase where samples are checked for inclusion - we source consent data from static database tables.

We can augment this by also allowing a fetch from something like CTRL.

This ticket would be all about the integration work with whatever system is chosen - and the data modelling of their consent, identifiers, service<->service auth etc

**Comment by andrewpatto** [comment]
Do we have a sense of the existing APIs on CTRL for fetching this info (per patient) - or will these APIs at the CTRL end need to be developed?

**Comment by DoxasticFox** [comment]
@andrewpatto There might already be a REST API which the front end uses. I'll investigate whether it's sufficient. I think the authentication might need to be changed, at least. It's currently only accessible for logged in users, and I think a user's "logged in" status depends on a cookie. But we'll need to send a token in a header or something.

**Comment by andrewpatto** [comment]
Yeah we'll definitely need some sort of service<->service auth story

**Comment by DoxasticFox** [comment]
I've spun-up a CTRL instance for testing this. Example invocation:

```bash
curl \
  http://34.151.86.85:3000/api/v1/duo_limitations \
  -s \
  -X POST \
  -H "Authorization: Bearer ${token}" \
  -d '
     [
       "non-existent-study-id",
       "A1543457"
     ]' \
  | jq
```
